src/v/cluster/scheduling/allocation_node.cc

```diff
@@ -15,11 +15,13 @@ namespace cluster {
 allocation_node::allocation_node(
   model::node_id id,
   uint32_t cpus,
-  absl::node_hash_map<ss::sstring, ss::sstring> labels)
+  absl::node_hash_map<ss::sstring, ss::sstring> labels,
+  std::optional<ss::sstring> rack)
   : _id(id)
   , _weights(cpus)
   , _max_capacity((cpus * max_allocations_per_core) - core0_extra_weight)
-  , _machine_labels(std::move(labels)) {
+  , _machine_labels(std::move(labels))
+  , _rack(std::move(rack)) {
```

Comment:
could you add a little context to the commit / cover letter about where the rack information comes from, where it's stored, etc...? presumably it is expressed in redpanda.yaml, but is it transmitted over the network by peers when they register with the control? I'm a little confused about how all that works.